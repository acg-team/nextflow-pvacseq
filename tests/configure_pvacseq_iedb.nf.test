nextflow_workflow {

    name "Test Subworkflow CONFIGURE_PVACSEQ_IEDB"
    script "subworkflows/local/configure_pvacseq_iedb/main.nf"
    workflow "CONFIGURE_PVACSEQ_IEDB"

    test("Only MHC Class I is required") {

        when {
            params {
                pvacseq_iedb_dir = "$projectDir/tests/tmp/iedb_i_only"
                pvacseq_algorithm = "NetMHCpan"
            }
            workflow {
                """
                input[0] = "$params.pvacseq_iedb_dir"
                input[1] = "$params.pvacseq_algorithm"
                """
            }
        }

        then {
            assert workflow.success
            assert snapshot(workflow.out).match()
        }
    }

    test("Only MHC Class II is required") {

        when {
            params {
                pvacseq_iedb_dir = "$projectDir/tests/tmp/iedb_ii_only"
                pvacseq_algorithm = "NetMHCIIpan"
            }
            workflow {
                """
                input[0] = "$params.pvacseq_iedb_dir"
                input[1] = "$params.pvacseq_algorithm"
                """
            }
        }

        then {
            assert workflow.success
            assert snapshot(workflow.out).match()
        }
    }

    test("Both MHC Class I and II are required") {

        when {
            params {
                pvacseq_iedb_dir = "$projectDir/tests/tmp/iedb_both"
                pvacseq_algorithm = "NetMHCpan NetMHCIIpan"
            }
            workflow {
                """
                input[0] = "$params.pvacseq_iedb_dir"
                input[1] = "$params.pvacseq_algorithm"
                """
            }
        }

        then {
            assert workflow.success
            assert snapshot(workflow.out).match()
        }
    }

    test("Invalid algorithm should fail") {

        when {
            params {
                pvacseq_iedb_dir = "$projectDir/tests/tmp/invalid"
                pvacseq_algorithm = "InvalidAlgo"
            }
            workflow {
                """
                input[0] = "$params.pvacseq_iedb_dir"
                input[1] = "$params.pvacseq_algorithm"
                """
            }
        }

        then {
            assert workflow.failed
        }
    }
}
