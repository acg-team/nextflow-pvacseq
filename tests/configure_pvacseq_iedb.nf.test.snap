{
    "Both MHC Class I and II are required": {
        "content": [
            207372,
            83505,
            123867
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.4"
        },
        "timestamp": "2025-08-31T14:17:05.052589454"
    },
    "Only MHC Class I is required": {
        "content": [
            83505,
            83505,
            0
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.4"
        },
        "timestamp": "2025-08-31T14:16:38.877401781"
    },
    "Only MHC Class II is required": {
        "content": [
            123867,
            0,
            123867
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.4"
        },
        "timestamp": "2025-08-31T14:16:50.454789846"
    }
}
